core/src/Streamly/Internal/Data/IORef.hs

```diff
@@ -88,20 +93,30 @@ modifyIORef' var g = do
   x <- readIORef var
   writeIORef var (g x)
 
--- | Generate a stream by continuously reading the IORef.
---
--- This operation reads the IORef without any synchronization. It can be
--- assumed to be atomic because the IORef (MutableByteArray) is always aligned
--- to Int boundaries, we are assuming that compiler uses single instructions to
--- access the memory. It may read stale values though until caches are
--- synchronised in a multiprocessor architecture.
---
--- /Pre-release/
-{-# INLINE_NORMAL pollIntIORef #-}
-pollIntIORef :: (MonadIO m, Unbox a) => IORef a -> D.Stream m a
-pollIntIORef var = D.Stream step ()
+-- | Internal, do not use.
+{-# INLINE_NORMAL pollGenericIORef #-}
+pollGenericIORef :: (MonadIO m, Unbox a) => IORef a -> D.Stream m a
```

Comment:
Why do we need a generic version? Just the `Int` version should be enough. Any acceptable generic value should be coerced to `Int`.

Reply:
To suppress the deprecation warning in the MutArray module. This definition is being used there.